src/librustc/ty/layout.rs

```diff
@@ -825,10 +825,18 @@ impl<'tcx> LayoutCx<'tcx, TyCtxt<'tcx>> {
                     });
                     (present_variants.next(), present_variants.next())
                 };
-                if present_first.is_none() {
-                    // Uninhabited because it has no variants, or only absent ones.
-                    return tcx.layout_raw(param_env.and(tcx.types.never));
-                }
+                let present_first = if present_first.is_none() {
```

Comment:
This would be cleaner as a `match` with a guard on `None if def.is_enum() =>` imo.